clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -3426,11 +3426,16 @@ Sema::InstantiateClass(SourceLocation PointOfInstantiation,
     return true;
 
   llvm::TimeTraceScope TimeScope("InstantiateClass", [&]() {
-    std::string Name;
-    llvm::raw_string_ostream OS(Name);
+    llvm::TimeTraceMetadata M;
+    llvm::raw_string_ostream OS(M.Detail);
     Instantiation->getNameForDiagnostic(OS, getPrintingPolicy(),
                                         /*Qualified=*/true);
-    return Name;
+    if (llvm::isTimeTraceVerbose()) {
```

Comment:
NIT: you might want to extract this into a function to share code.
Something like `RecordLocationInMetadata`.

It's not a big deal, though, with just few uses.

Reply:
Resolving duplicate comment.